### PR TITLE
[wasm] Properly handle trailing whitespace in the URL for the HttpClient

### LINKED ISSUE #70025: Blazor wasm HttpClient ignores spaces at the end of URIs even if it is escaped (%20)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Using HttpClient from Blazor wasm to send an HTTP request with a URI that ends with a space results in the space at the end being ignored and not sent as part of url of the actual request the browser sends. E.g. the following code:
```
await HttpClient.GetAsync("/api/test%20");
```
Incorrectly results in a request to `GET /api/test` (with no space at the end)

However, escaping spaces in other parts of the URI works just fine. E.g.:
```
await HttpClient.GetAsync("/api/test%20test");
```
Correctly results in a request to `GET /api/test test`

Also, as a workaround, an empty query can be added, and it works correctly:
```
await HttpClient.GetAsync("/api/test%20?");
```
Correctly results in a request to `GET /api/test ` (with space)

The same behavior is exhibited when using the overloads that take a `Uri` instead of `string`.

### Expected Behavior

`HttpClient.GetAsync("/api/test%20")` should result in a request to `GET /api/test%20` not `GET /api/test`

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

6.0.202

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by maraf**
- If the space is escaped, it remains in the URL.
- If the space is not escaped, it gets trimmed.

Fixes https://github.com/dotnet/runtime/issues/70025

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Url.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        public async Task TrailingWhitespace_Escaped()
+        {
+            string path = await RunGetAndReturnServerPathAsync("/test%20");
+            Assert.Equal("/test%20", path);
+        }
+
+        [Fact]
+        public async Task TrailingWhitespace_NotEscaped()
+        {
+            string path = await RunGetAndReturnServerPathAsync("/test ");
```
Comment (radical) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Url.cs:51]: What happens if you have a query string too? like `/test ?a=b`?

Comment (maraf) [reply src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Url.cs:51]: I have added tests.

`/test ?a=b` => `/test%20?a=b`
`/test%20?a=b` => `/test%20?a=b`

Same as in javascript fetch.

**Review by pavelsavara — APPROVED** [review]

**Comment by maraf** [comment]
Failures are not relevant.

**Pull request merged**